packages/terra-search-field/tests/wdio/search-field-spec.js

```diff
@@ -0,0 +1,252 @@
+
+    it('should enter a long enough search term', () => {
+      browser.addValue('input', 'm');
+    });
+
+    Terra.should.matchScreenshot('with long enough text');
+  });
+
+  describe('Callback', () => {
+    before(() => browser.url('/#/raw/tests/terra-search-field/search-field/callback-search-field'));
+
+    Terra.should.matchScreenshot('empty');
+
+    it('should enter a short search term', () => {
+      browser.setValue('input', 'Lo');
+      browser.execute(() => {
+        // Removes the blinking cursor to prevent screenshot mismatches.
+        document.querySelector('input').style.caretColor = 'transparent';
+      });
+    });
+
+    Terra.should.matchScreenshot('with too short text');
+
+    it('should enter a short search term', () => {
+      browser.addValue('input', 'rem');
+    });
+
+    Terra.should.matchScreenshot('with long enough text');
+  });
+
+  describe('On Change', () => {
+    before(() => browser.url('/#/raw/tests/terra-search-field/search-field/search-field-on-change'));
+
+    it('should enter a letter', () => {
+      browser.setValue('input', 'L');
+      browser.execute(() => {
+        // Removes the blinking cursor to prevent screenshot mismatches.
+        document.querySelector('input').style.caretColor = 'transparent';
+      });
+    });
+
+    Terra.should.matchScreenshot('updated once');
+
+    it('should enter another letter', () => {
+      browser.addValue('input', 'o');
+    });
+
+    Terra.should.matchScreenshot('updated twice');
+  });
+
+  describe('Search With Enter', () => {
+    before(() => browser.url('/#/raw/tests/terra-search-field/search-field/search-field-enter'));
+
+    it('should search with enter', () => {
+      browser.click('input');
+      browser.keys('Enter');
+      browser.execute(() => {
+        // Removes the blinking cursor to prevent screenshot mismatches.
+        document.querySelector('input').style.caretColor = 'transparent';
+      });
+    });
+
+    Terra.should.matchScreenshot();
+  });
+
+  describe('Auto Search Disabled', () => {
+    before(() => browser.url('/#/raw/tests/terra-search-field/search-field/auto-search-disabled-search-field'));
+
+    it('should enter a search term', () => {
+      browser.setValue('input', 'Lorem');
+      browser.execute(() => {
+        // Removes the blinking cursor to prevent screenshot mismatches.
+        document.querySelector('input').style.caretColor = 'transparent';
+      });
+    });
+
+    Terra.should.matchScreenshot('text before search');
+
+    it('should search with the button', () => {
+      browser.click('button');
+      // Ensure button on hover styling is disabled
+      browser.click('h3');
+    });
+
+    Terra.should.matchScreenshot('searched text');
+
+    it('should search using enter', () => {
+      browser.addValue('input', ' ipsum');
+      browser.keys('Enter');
+    });
+
+    Terra.should.matchScreenshot('extended search');
+  });
+
+  describe('Delayed', () => {
```

Comment:
Nit: update to AutoSearch Delayed

Then, we should add non-screenshot assertion here every 250 ms like nightwatch did to ensure the search was not yet triggered. A screenshot test overly appropriate here because we are testing the callback. It might make sense to move this to jest.

Reply:
Ended up moving this to jest, but left the test page.